wicket-core/src/main/java/org/apache/wicket/ajax/AbstractDefaultAjaxBehavior.java

```diff
@@ -83,12 +83,12 @@ protected void onBind()
 		final Component component = getComponent();
 		
 		component.setOutputMarkupId(true);
-		
-		if (getStatelessHint(component))
-		{
-			//generate behavior id
-			component.getBehaviorId(this);
-		}
+	}
+	
+	@Override
```

Comment:
Hi,
forcing Ajax behavior to be stateful by default will break the current logic for stateless Ajax component. See [user guide](https://ci.apache.org/projects/wicket/guide/8.x/single.html#_stateless_ajax_components_behaviors). 
I would remove this isStableIdRequired override and just keep things as they are now, i.e. behaviors state (stateful/stateless) depends on container component.

Reply:
`isStableIdRequired` does not make the behavior stateful, that's the purpose of this new method.

Reply:
ok. thanks for the explanation!